### ISSUE TITLE
[BUG] grp.getgrall() in user.py causing performance issue during minion connections

### ISSUE THREAD

**Issue by wcbzero** [saltstack/salt#64888]
**Description**
On systems with a large number of non-local groups, grp.getgrall() is an expensive operation. When large numbers of minions connect this can cause performance issues, and minion connection timeouts

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info. There is no general set-up of Salt.)

Please be as specific as possible and give set-up details.

- on-prem machine
- VM (VMWare ESXi)
- OpenLDAP Slapd LDAP Caching Server
- Active Directory with > 1000 groups

**Steps to Reproduce the behavior**

1. Configure Slapd as a caching server for an Active Directory instance containing more than 1000 groups with Unix GID numbers.
2. Start the salt master
3. Have > 500 Machines connect to the machine
4. Observe large number of slow grp.getgrall() calls

**Expected behavior**

Only lookups of the groups the user is a member of happen.

**Screenshots**

**Versions Report**
<details><summary>
```Salt Version:
          Salt: 3006.1

Python Version:
        Python: 3.8.5 (default, May 27 2021, 13:30:53)

Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: Not Installed
     docker-py: 5.0.3
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.2.0
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.4
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 23.1
     pycparser: Not Installed
      pycrypto: 3.15.0
  pycryptodome: 3.12.0
        pygit2: Not Installed
  python-gnupg: Not Installed
        PyYAML: 6.0
         PyZMQ: 25.0.2
        relenv: Not Installed
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: ubuntu 20.04.6 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-148-generic
        system: Linux
       version: Ubuntu 20.04.6 focal</summary>
```

</details>

**Comment by wcbzero** [comment]
I created a Patch in PR #64891

**Comment by whytewolf** [comment]
looking into this and https://github.com/saltstack/salt/issues/64953 to try and determine the best coarse of action. right now it is looking like a revert of the code in question. but i want to talk to @nicholasmhughes first. this does not look like it will be in 3006.3.

**Comment by nicholasmhughes** [comment]
Fixed by #65077
